ambry-server/src/test/java/com.github.ambry.server/Http2BlockingChannel.java

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.network.ChannelOutput;
+import com.github.ambry.network.ConnectedChannel;
+import com.github.ambry.network.Send;
+import com.github.ambry.rest.NettySslHttp2Factory;
+import com.github.ambry.utils.ByteBufferChannel;
+import com.github.ambry.utils.NettyByteBufDataInputStream;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http2.DefaultHttp2DataFrame;
+import io.netty.handler.codec.http2.DefaultHttp2Headers;
+import io.netty.handler.codec.http2.DefaultHttp2HeadersFrame;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.handler.codec.http2.Http2StreamChannel;
+import io.netty.handler.codec.http2.Http2StreamChannelBootstrap;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A HTTP2 implementation of {@link ConnectedChannel}
+ */
+public class Http2BlockingChannel implements ConnectedChannel {
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+  private Channel channel;
+  private Http2ResponseHandler http2ResponseHandler;
+  private String hostName;
+  private int port;
+
+  public Http2BlockingChannel(String hostName, int port) {
+    http2ResponseHandler = new Http2ResponseHandler();
+    this.hostName = hostName;
+    this.port = port;
+  }
+
+  @Override
+  public void connect() throws IOException {
+    Bootstrap b = new Bootstrap();
+    b.group(new NioEventLoopGroup());
+    b.channel(NioSocketChannel.class);
+    b.option(ChannelOption.SO_KEEPALIVE, true);
+    b.remoteAddress(hostName, port);
+    SSLFactory sslFactory;
+    try {
+      sslFactory = new NettySslHttp2Factory(new SSLConfig(new VerifiableProperties(new Properties())));
+    } catch (GeneralSecurityException e) {
+      throw new IOException(e);
+    }
+    b.handler(new Http2ClientChannelInitializer(sslFactory, hostName, port));
+
+    // Start the client.
+    channel = b.connect().syncUninterruptibly().channel();
+    logger.info("Connected to remote host");
+  }
+
+  @Override
+  public void disconnect() throws IOException {
+
```

Comment:
does this need to be implemented? For example, shutting down the EventLoopGroup?